docs/aws/audit/redshiftmonitoring/rules/redshift_desired_node_type_remediation.mdx

```diff
@@ -1,6 +1,222 @@
+3. **Define Desired Node Type**: Specify the desired node type (`DESIRED_NODE_TYPE`).
+4. **Python Script**: The script:
+   - Retrieves all Redshift clusters.
+   - Checks if each cluster's node type matches the desired type.
+   - Modifies the node type if it does not match the desired type.
+
+This script ensures that all Redshift clusters are using the specified node type, thus preventing misconfigurations.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Services" from the top menu, then selecting "Redshift" under the "Database" category.
+3. In the Redshift dashboard, select "Clusters" from the left-hand menu. This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Node Type" column. This will display the type of nodes that the cluster is using. Compare this to your desired node type to see if there is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all the Redshift clusters in your account using the following command:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier]' --output text
+   ```
+
+   This command will return a list of all Redshift cluster identifiers.
+
+3. Now, for each cluster identifier, you can describe the cluster to get the node type. Use the following command:
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <cluster-identifier> --query 'Clusters[*].[NodeType]' --output text
+   ```
+
+   Replace `<cluster-identifier>` with the actual cluster identifier. This command will return the node type of the specified Redshift cluster.
+
+4. Finally, compare the returned node type with your desired node type. If they don't match, then the Redshift cluster node is not of the desired type. You can automate this process using a Python script or a shell script to loop through all the clusters and compare their node types with the desired node type.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This SDK allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create an AWS session using your credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>